Source/WebCore/platform/graphics/texmap/coordinated/GraphicsLayerCoordinated.cpp

```diff
@@ -101,17 +109,16 @@ void GraphicsLayerCoordinated::setNeedsDisplayInRect(const FloatRect& initialRec
     if (rect.isEmpty())
         return;
 
-    auto& rects = m_dirtyRegion.rects;
-    bool alreadyRecorded = std::any_of(rects.begin(), rects.end(), [&](auto& dirtyRect) {
-        return dirtyRect.contains(rect);
-    });
-    if (alreadyRecorded)
-        return;
+    addRepaintRect(rect);
 
-    rects.append(rect);
-    noteLayerPropertyChanged(Change::DirtyRegion, ScheduleFlush::Yes);
+    if (!m_dirtyRegion) {
+        static constexpr FloatSize minDamageSize = { 512, 512 };
+        auto viewSize = client().viewSize();
+        m_dirtyRegion = Damage(m_size.constrainedBetween(minDamageSize, viewSize));
```

Comment:
Doesn't that imply dirty regions are always at least 512x512 px wide? Basically always covering the whole tile (when default tile size is used?). That seems wrong.